framework/src/vectorpostprocessors/PointSamplerBase.C

```diff
@@ -80,6 +98,31 @@ PointSamplerBase::execute()
 void
 PointSamplerBase::finalize()
 {
+  // Save off for speed
+  unsigned int pid = processor_id();
+
+  /*
+   * Figure out which processor is actually going "claim" each point.
+   * If multiple processors found the point and computed values what happens is that
+   * maxloc will give us the smallest PID in max_id
+   */
+  std::vector<unsigned int> max_id(_found_points.size());
+
+  _communicator.maxloc(_found_points, max_id);
+
+  for (unsigned int i=0; i<max_id.size(); i++)
+  {
+    // Only do this check on the proc zero because it's the same on every processor
+    // _found_points should contain all 1's at this point (ie every point was found by a proc)
+    if (pid == 0 && !_found_points[i])
+      mooseError("Sample point not found: " << _points[i]);
+
+    unsigned int id = max_id[i];
+
+    if (id == pid)
```

Comment:
Heh - yes it could.  I saw that too once I was looking at the diff on GitHub.  I think I had grander plans for `id` :smile: 

I'll change it